docs/src/v14_migration_guide.md

```diff
@@ -0,0 +1,193 @@
+# Version 14 Migration Guide
```

Comment:
I think these are the points we should mention, but tell me if I forgot something.